bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,347 @@
++ADB_PORT=5555
++USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
+ 
+ #---------      Functions    -------------------
+ function error() {
+@@ -575,6 +580,26 @@ function parse_arg() {
+                 setup_remote_infer || return -1
+                 ;;
+ 
++	    -s)
++		echo "Skip initial setup"
++                skip_initial_host_setup=true
++		;;
++
++	    -v)
++                VSOCK_ID=$2
++		shift
++		;;
++
++	    -b)
++                ADB_PORT=$2
++		shift
++                ;;
++
++            -n)
++                VM_NAME=$2
++		shift
++                ;;
++
+             --auto-start)
+                 install_auto_start_service "$2" || return -1
+                 shift
+@@ -599,13 +624,26 @@ function parse_arg() {
+ }
+ 
+ function setup_civ_ini() {
+-        $CIV_WORK_DIR/scripts/setup_civ_ini.sh
++        $CIV_WORK_DIR/scripts/setup_civ_ini.sh -v $1 -p $2 -n $3
+ }
+ 
+ function create_aaf_dir() {
+         mkdir -p $CIV_WORK_DIR/scripts/aaf
+ }
+ 
++function create_vm_dir() {
++	echo "Dir: $CIV_WORK_DIR/$VM_NAME"
++	mkdir $CIV_WORK_DIR/$VM_NAME
++}
++
++function copy_files_for_vm() {
++    echo "Copying file: $CIV_WORK_DIR/$VM_NAME"
++    mkdir $CIV_WORK_DIR/$VM_NAME/scripts
++    cp -r $CIV_WORK_DIR/scripts/aaf $CIV_WORK_DIR/$VM_NAME/scripts/
++    cp OVMF.fd $CIV_WORK_DIR/$VM_NAME/
++    ls -la $CIV_WORK_DIR/$VM_NAME/
++    cp $CIV_WORK_DIR/scripts/rpmb_dev $CIV_WORK_DIR/$VM_NAME/scripts/
++}
+ 
+ #-------------    main processes    -------------
+ 
+@@ -618,22 +656,21 @@ check_network
+ check_kernel_version
+ check_sriov_setup
+ 
+-ubu_changes_require
+-ubu_install_qemu_gvt
+-ubu_build_ovmf_gvt
+-ubu_enable_host_gvt
+-ubu_update_fw
++if [ $skip_initial_host_setup != true ]; then
++    ubu_changes_require
++    ubu_install_qemu_gvt
++    ubu_enable_host_gvt
++    ubu_update_fw
++    ubu_build_ovmf_gvt
++fi
+ 
+-install_vm_manager
++create_vm_dir
+ create_aaf_dir
+-setup_civ_ini
+-
+-prepare_required_scripts
+-ubu_install_swtpm
+-ubu_install_libssl
+-ubu_update_bt_fw
+-set_sleep_inhibitor
++setup_civ_ini $VSOCK_ID $ADB_PORT $VM_NAME
++copy_files_for_vm
+ 
+-ask_reboot
++if [ $skip_initial_host_setup != true ]; then
++    ask_reboot
++fi
+ 
+ echo "Done: \"$(realpath $0) $@\""
+diff --git a/scripts/setup_host_multiple_android_vm.sh b/scripts/setup_host_multiple_android_vm.sh
+new file mode 100755
+index 0000000..894de87
+--- /dev/null
++++ b/scripts/setup_host_multiple_android_vm.sh
+@@ -0,0 +1,80 @@
++#!/bin/bash
++
++# Copyright (c) 2020 Intel Corporation.
++# All rights reserved.
++#
++# SPDX-License-Identifier: Apache-2.0
++
++set -eE
++
++#---------      Global variable     -------------------
++CIV_DIR=$(pwd)
++USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
++VM_NAME=Android_CIV
++VSOCK_ID=3
++NUMBER_OF_VM=1
++ADB_PORT=5555
++INITIAL_VM_NUM=1
++SETUP_HOST_DONE=false
++
++function setup_files() {
++    echo "Creating setup for VMs"
++    if [ -e $CIV_DIR/edk2 ]
++    then
++        SETUP_HOST_DONE=true
```

Comment:
I don't think checking edk2 folder is a good way of checking that setup_host is done since that is a build folder.
Maybe check the initial folder structure created by setup_host.sh for the first VM?